bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/api/TapoDeviceHttpApi.java

```diff
@@ -411,6 +418,26 @@ protected Integer getErrorCode(String responseBody) {
         }
     }
 
+    /**
+     * Check for JsonObject "errorcode" and if this is > 0 (no Error)
+     * 
+     * @param responseBody
+     * @return true if is js errorcode > 0; false if there is no "errorcode"
+     */
+    protected Boolean hasErrorCode(String responseBody) {
+        try {
+            JsonObject jsonObject = gson.fromJson(responseBody, JsonObject.class);
+            /* get errocode (0=success) */
+            Integer errorCode = jsonObjectToInt(jsonObject, "error_code", ERR_JSON_DECODE_FAIL);
+            if (errorCode > 0) {
+                return true;
+            }
+        } catch (Exception e) {
```

Comment:
Please catch more precise exceptions, those that are thrown by gson.fromJson and jsonObjectToInt

Reply:
Just need value of Json-Field "errorCode" if exists. sometimes string is a decryptet non json text. Now im checking if it's json before